File: Chain of Responsibility/Ruby/Authorization.rb
require_relative "Handler"
require_relative "Response"

class Authorization < Handler
    def execute(request)
        if request.authorized
            puts "Authorized"
            return @next_handler.execute(request)
        else
            return Response.new(
                code: 403, 
                message: "Unauthorized",
                data: nil
            )
        end
    end
end

File: Chain of Responsibility/Javascript/Authorization.js
import Handler from "./Handler.js";
import Response from "./Response.js";

class Authorization extends Handler {
  execute(request) {
    if (request.authorized === true) {
      console.log("Authorized");

      return this.next_handler.execute(request);
    } else {
      return new Response(403, "Unauthorized", null);
    }
  }
}

export default Authorization;
